### ISSUE TITLE
F1 sending people to StackExchange feels… weird

### ISSUE THREAD

**Issue by cassidyjames** [elementary/files#1475]
I am not a huge fan of undiscoverable keyboard shortcuts in general, but I especially think we should **not** be throwing open a web browser window to a third-party site when someone hits a keyboard key. If we want something similar, I think a buffer dialog would make a bit more sense, but I'm not sure how practical an F1 help key is, anyway—it feels like a really old pattern that isn't consistently used in apps these days.

**Comment by jeremypw** [comment]
I also do not like the link to a third-party web-site although I would point out that the elementary web-site redirects people to StackExchange in more than one place for help on the native apps.

I disagree that the `F1` key is necessarily impractical and that fact that it is an old pattern does not mean it is not useful - unless you think keyboard shortcuts in general are not useful (which they are not I guess for phones and tablets).  I can see that having two keys for shortcut help (The Super key shows *system* shortcuts) is not optimal. The solution to that would be for the "Super" key to show that shortcuts for the current app as well - but that is technically difficult. I don't think having a different shortcut for app help is too confusing.

If you veto both a shortcut and an appmenu for accessing help then you are pretty much out of options.

What is weird is that a distribution that claims to be easy to use makes it hard to find out what the keyboard shortcuts for an app are - especially for users who cannot or do not use a mouse.

**Comment by cassidyjames** [comment]
@jeremypw we have been moving to more contextual disclosure of shortcuts in tooltips and menus, and that's generally preferred over a "shortcut sheet" for a few reasons:

- Presenting it contextually means we're directly associating it with the action that will be performed.
- Contextual leads to a more naturally _learnable_ interface instead of just requiring memorization.
- You're more likely to discover a contextual keyboard shortcut because it doesn't require actively looking for all shortcuts.

Of course with more abstract actions that don't have an equivalent in the UI (like some OS-wide shortcuts), a shortcut sheet can be handy. But I would really lean toward ensuring we're disclosing things contextually in all the right places, first. 

Not being able to use the mouse is a good point, and could lean on a shortcut sheet for that case. But that still doesn't mean to throw people to StackExchange on a keypress. :)

**Comment by hanaral** [comment]
Optimally there would be akey you can press that would display all the hover tooltips at once withtout somehow making it look cluttered.

**Comment by jeremypw** [comment]
Yes, this is about people who mainly or only use the keyboard.  They cannot see tooltips for example and opening menus can be awkward.  They are the ones who need to know the keyboard shortcuts. I think we are agreed that StackExchange is not a good solution (and does not generally give shortcut info anyway).  Now we just need a solution that can consistently be applied across the native apps.

**Comment by jeremypw** [comment]
There is a debate to be had as to whethere any *other* kind of help is useful.  I guess the aim is to make the apps self-explanatory and not need any tutorials.  However there are quite a few "how do I?" questions about Files on StackExchange.

**Comment by jeremypw** [comment]
Still occurring in OS8.  Needs to be addressed. Other elementary apps like Code and Terminal do not do this so can probably be discontinued.
